### DISCUSSION TITLE
device descriptor request failed

### DISCUSSION THREAD

**Discussion by arilink-tech** [hathach/tinyusb#696] [General]
**Set up**
[Mandatory] Provide details of your setup help us to reproduce the issue as quick as possible  
 - **PC OS**   :Windows 10
 - **Board**   : STM32 F407VGXX
 - **Firmware**: examples/device/hid_composite_freeRTOS

**Describe the bug**
0.Projects created using the master branch on 2.24 can run normally
1.tusb_init() success  
2.The windows device manager prompts that the device descriptor request failed
3.It looks like uint8_t const * tud_descriptor_device_cb(void) function is not executed

**Screenshots**
If applicable, add screenshots, bus capture to help explain your problem. 
![image](https://user-images.githubusercontent.com/30901123/109803798-71063600-7c5c-11eb-8bd9-4240c1ac2fe2.png)

**Comment by arilink-tech** [comment]
I tested it in depth and found that the problem appeared in the loop of tud_task(), so I added log output to the tud_task() function, and found that the program can enter the tud_task()->while(1) loop and be able to get from the queue I received the event, but it is strange that the DCD_EVENT_BUS_RESET event can be received, but the DCD_EVENT_SETUP_RECEIVED cannot be received. This is why the process_control_request() cannot be executed and cannot respond to the host's descriptor request, which causes windows to prompt me that the device descriptor request failed. Knowing the above but not sure about the next operation, can you please help me?

**Comment by arilink-tech** [comment]
![image](https://user-images.githubusercontent.com/30901123/109922525-6f8a4b80-7cf8-11eb-9a37-2d71addbf63b.png)

**Comment by hathach** [comment]
I am not familiar with your IDE and set up. Look like you are having some integration issue which is not really qualified as stack bug. Which I have no idea how to fix. I will move it to discussion, maybe others with similar ide set up could help

**Reply by arilink-tech** [reply]
This problem has been solved. I have a problem with my hardware.  Thank you very much
